### DISCUSSION TITLE
Can I make spack choose libstdc++ dependency?

### DISCUSSION THREAD

**Discussion by ipcamit** [spack/spack#36007] [General]
I am trying to install llvm in an LLNL HPC. I get the following error:
```
  >> 56    CMake Error at cmake/modules/CheckCompilerVersion.cmake:120 (message):
     57      libstdc++ version should be at least 5.1 because LLVM will soon use new C++
     58      features which your toolchain version doesn't support.  You can temporarily
     59      opt out using LLVM_TEMPORARILY_A
```
I believe this is because clang is still using older c++ toolchain:

```
$ clang -x c++ -E -v -
clang version 14.0.4
...
 /usr/tce/packages/gcc/gcc-4.9.3/lib64/gcc/x86_64-unknown-linux-gnu/4.9.3/../../../../include/c++/4.9.3
 ```

Can I make spack install its own version updated version of libstd++? Usually we resolve this by passing `--gcc-toolchain=/opt/rh/devtoolset-10/root` flag to the clang compilers.

I tried explicitly providing the gcc dependency, but to no avail
```
$ spack spec llvm@14.0.6 %clang ^gcc@9.3.1
==> Error: Package llvm does not depend on gcc@9.3.1
```

**Comment by climbfuji** [comment]
See lines 16-18 in this example: https://github.com/NOAA-EMC/spack-stack/blob/develop/configs/sites/hera/compilers.yaml (this is one way, there are other ways described in detail in the spack documentation https://spack.readthedocs.io/en/latest/).
